autoscale/src/main/java/com/sequenceiq/periscope/service/YarnInvocationService.java

```diff
@@ -0,0 +1,138 @@
+
+        Optional<Integer> mandatoryDownScaleCount = Optional.of(maxResourceValueOffset)
+                .filter(mandatoryDownscale -> mandatoryDownscale < 0).map(downscale -> -1 * downscale);
+
+        YarnScalingServiceV1Response yarnResponse =
+                yarnMetricsClient.getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, pollingUserCrn, mandatoryDownScaleCount);
+
+        if (cluster.getUpdateFailedDetails() != null && pollingUserCrn.equals(cluster.getMachineUserCrn())) {
+            // Successful YARN API call
+            clusterService.setUpdateFailedDetails(cluster.getId(), null);
+        }
+
+        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup, maxAllowedUpScale,
+                loadAlertConfiguration.getMaxScaleUpStepSize());
+        List<String> yarnRecommendedDecommissionHosts =
+                yarnResponseUtils.getYarnRecommendedDecommissionHostsForHostGroup(yarnResponse, hostFqdnsToInstanceId,
+                        maxAllowedDownScale, mandatoryDownScaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
+
+        LOGGER.info("yarnRecommendedScaleUpCount={}, yarnRecommendedDecommission={}", yarnRecommendedScaleUpCount, yarnRecommendedDecommissionHosts);
+
+        ScalingAdjustmentType adjustmentType = !stopStartEnabled ? REGULAR : STOPSTART;
+
+        publishScalingEventIfNeeded(yarnRecommendedScaleUpCount, stackV4Response, existingHostGroupSize, serviceHealthyHostGroupSize,
+                yarnRecommendedDecommissionHosts, stoppedHostInstanceIds, adjustmentType);
+    }
+
+    protected boolean isCoolDownTimeElapsed(String clusterCrn, String coolDownAction, long expectedCoolDownMillis, long lastClusterScalingActivity) {
+        long remainingTime = ClusterUtils.getRemainingCooldownTime(
+                expectedCoolDownMillis, lastClusterScalingActivity);
+
+        if (remainingTime <= 0) {
+            return true;
+        } else {
+            LOGGER.debug("Cluster {} cannot be {} for {} min(s)", clusterCrn, coolDownAction,
+                    ClusterUtils.TIME_FORMAT.format((double) remainingTime / TimeUtil.MIN_IN_MS));
+        }
+        return false;
+    }
+
+    private void publishScalingEventIfNeeded(int yarnRecommendedScaleUpCount, StackV4Response stackV4Response, int existingHostGroupSize,
+            int serviceHealthyHostGroupSize, List<String> yarnRecommendedDecommissionHosts, List<String> stoppedHosts, ScalingAdjustmentType adjustmentType) {
+        if (yarnRecommendedScaleUpCount > 0 && isCoolDownTimeElapsed(cluster.getStackCrn(), "scaled-up",
+                loadAlertConfiguration.getScaleUpCoolDownMillis(), cluster.getLastScalingActivity()))  {
+            if (STOPSTART.equals(adjustmentType)) {
+                Integer existingClusterNodeCount = stackV4Response.getNodeCount() - stoppedHosts.size();
```

Comment:
Please create a separate jira to fix this so that ScalingEvent has existClusterNodeCount and stoppedNodeCount as separate parameters.
Wherever this is used: should do the relevant math (The parameter name vs actual value will otherwise end up being very confusing for anyone reading the code)

Reply:
Done. https://jira.cloudera.com/browse/CB-18606